proto/structured.proto

```diff
@@ -104,3 +105,234 @@ message CreateTableResponse {
+// of type integer. Incrementing by 0 is not a noop, and create a zero value.
+message IncrementTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ColumnCollection row = 2 [(gogoproto.nullable) = false];
+}
+
+// An IncrementTableResponse returns the row with the new column values.
+message IncrementTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ValueCollection row = 2 [(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowRequest deletes the entire row. key contains the columns
+// forming the index key.
+message DeleteTableRowRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional ColumnCollection key = 2 [(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowResponse.
+message DeleteTableRowResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// A DeleteTableRowRangeRequest deletes the specified rows. Both key and end_key
+// must only contain the columns forming the index key.
+message DeleteTableRowRangeRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Delete *all* entries between Key (inclusive) and EndKey
+  // (exclusive). Must be > 0
+  optional ColumnCollection key = 2 [(gogoproto.nullable) = false];
+  optional ColumnCollection end_key = 3 [(gogoproto.nullable) = false];
+  optional int64 max_entries_to_delete = 4 [(gogoproto.nullable) = false];
+}
+
+// A DeleteTableRowRangeResponse.
+message DeleteTableRowRangeResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Number of entries deleted.
+  optional int64 num_deleted = 2 [(gogoproto.nullable) = false];
+}
+
+// A ScanTableRequest. It specifies the start and end keys for the scan
+// and the maximum number of results.
+message ScanTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Both key and end_key must only contain the columns forming the index key.
+  optional ColumnCollection key = 3 [(gogoproto.nullable) = false];
+  optional ColumnCollection end_key = 4 [(gogoproto.nullable) = false];
+  optional int64 max_results = 5 [(gogoproto.nullable) = false];
+  // get specified columns from the scaned rows.
+  repeated string columns = 6;
+}
+
+// A ScanTableResponse is the response.
+message ScanTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // Empty in case of error.
+  repeated ValueCollection rows = 2;
+}
+
+// A BatchTableRequest contains one or more requests to be executed in
+// parallel, or if applicable (based on write-only commands and
+// range-locality), as a single update.
+//
+message BatchTableRequest {
```

Comment:
A separate `BatchTableRequest` means we won't be able to mix key/value and structured operations which might affect the migration. Perhaps `BatchTable{Request,Response}` should contain a `Batch{Request,Response}` for a transitionary period.